### ISSUE TITLE
false error when casting to a type with new union syntax

### ISSUE THREAD

**Issue by KotlinIsland** [python/mypy#11502]
```py
from typing import cast
cast(int | None, None)  # error: Cast target is not a type  [misc]
```

**Comment by sobolevn** [comment]
Please, attach a python version, when dealing with this rather new python type syntax 🙂 

It works for me on 3.10 and `master`:

```python
(.venv) ~/Desktop/mypy  master ✔                                                     
» mypy out/ex.py --show-traceback
out/ex.py:2: note: Revealed type is "Union[builtins.int, None]"

(.venv) ~/Desktop/mypy  master ✔                                                  1 ⚠️
» python --version 
Python 3.10.0
```
